portal/models/app_text.py

```diff
@@ -416,3 +419,14 @@ def app_text(name, *args):
         raise ValueError(
             "AppText with name '{}' defines more parameters "
             "than provided: `{}`".format(name, *args))
+
+
+def format_asset_attributes(asset, variables):
+    asset_out = ""
+    x = 0
+    for match in finditer(r"\${[^}]+}", asset):
```

Comment:
This feels like a reinvention of string format.  Can we not simply leverage that?  Something like:
```
asset_out = asset.format(**variables)
```